lua/lspsaga/rename.lua

```diff
@@ -31,20 +32,15 @@ end
 
 function rename:apply_action_keys()
   local modes = { 'i', 'n', 'v' }
+  local opts = { buffer = self.bufnr }
 
-  for i, mode in pairs(modes) do
-    if string.lower(config.rename.quit) ~= '<esc>' or mode == 'n' then
```

Comment:
I'm not sure why `<esc>` was only mapped in `normal` mode without the user knowing about it. I know that this is to get out of `insert` mode for instance, but if the user has explicitly assigned the `quit` action to `<Esc>` I don't think it should be only mapped to `n` but in all modes.

Reply:
make sense. and can we wrap own keymap function based on `nvim_buf_set_keymap` . `vim.keymap.set` is a syntax suger based on that and `nvim_set_keymap` . mostly we just set buffer map . in `ref` branch i changed some keymap by using `nvim_buf_set_keymap` you can take a look at it :)  and Thanks for this pr 👍

Reply:
I think `vim.keymap.set` provides a good abstraction over the internal implementation details of setting a keymap. But if you're mostly using `nvim_buf_set_keymap` and want to be explicit about that, then sure I can make the change!

Reply:
For lspsaga, most floating window keymaps are buffe rkeymaps. `vim.keymap.set` has some redundant wrapping and if else. It's not necessary for us so I replaced part of it in ref branch. So I'm thinking that we can simply wrap a function with `nvim_buf_set_keymap`. Replaces vim.keymap.set.

Reply:
I made the change and now `map_keys` uses `nvim_buf_set_keymap` by default.